xmake/toolchains/gcc/xmake.lua

```diff
@@ -62,6 +63,36 @@ toolchain("gcc" .. suffix)
             toolchain:add("ldflags", march)
             toolchain:add("shflags", march)
         end
+
+        local host_arch = os.arch()
+        local target_arch = toolchain:arch()
+
+        if host_arch == target_arch then
+            -- Early exit: prevents further configuration of this toolchain
+            return
+        elseif option.get("verbose") then
+            cprint("${bright yellow}cross compiling from %s to %s", host_arch, target_arch)
+        end
+
+        local target
+        if toolchain:is_arch("x86_64", "x64") then
+            target = "x86_64"
+        elseif toolchain:is_arch("i386", "x86", "i686") then
+            target = "i686"
+        elseif toolchain:is_arch("arm64", "aarch64") then
+            target = "aarch64"
+        elseif toolchain:is_arch("arm") then
+            target = "armv7"
+        end
+
+        -- TODO: Add support for more platforms, such as mingw.
+        if target and toolchain:is_plat("linux") then
+            target = target .. "-linux-gnu-"
```

Comment:
The gcc toolchain is only designed to support the host platform. Please use the cross toolchain.

it will break many things. e.g. packages, toolchain cross state, ...

https://github.com/xmake-io/xmake/blob/888f5c3b5452b2b8c6c47e89028c4449a238e31f/xmake/modules/private/utils/toolchain.lua#L28

https://github.com/xmake-io/xmake/blob/888f5c3b5452b2b8c6c47e89028c4449a238e31f/xmake/modules/package/tools/cmake.lua#L99